### ISSUE TITLE
False Positives on unused parameters in the constructor

### ISSUE THREAD

**Issue by iltar** [kalessil/phpinspectionsea#157]
When using a trait, it seems that the detection of unused properties in a constructor gives a few false positives. I've tried to reproduce this, but I was unable to do so for some reason, I can't seem to find the cause of it in my existing files either.

![image](https://cloud.githubusercontent.com/assets/1754678/23401348/a6f66498-fda7-11e6-815c-39c9b7e84d56.png)

![image](https://cloud.githubusercontent.com/assets/1754678/23401365/b8c8c92c-fda7-11e6-91c1-af1cf4d52586.png)

![image](https://cloud.githubusercontent.com/assets/1754678/23401397/d787da2e-fda7-11e6-8188-dbc2aeb315be.png)

**Comment by funivan** [comment]
Thins inspection contains some bugs. Here is my testCase =)
```php
    class UserCustomClass {
      private $name;
      private $nameLen;
      public function __construct(string $name) {
        $this->name = $name;
        $this->nameLen = strlen($name); //Error Here
      }
      public function getName(){
        return $this->name;
      }
    }

    class UserCustomClass {
      private $name;
      public function __construct(string $name) {
        $this->name = $name; // Expect error. But inspection does not fire
      }
    }

```

**Comment by kalessil** [comment]
Yep, confirm.

**Comment by kalessil** [comment]
@iltar, @funivan : fixed the both cases from you
